### ISSUE TITLE
[Feature Request]: `SplitPanel` component - `header` prop accept `React.Node`

### ISSUE THREAD

**Issue by nichita-pasecinic** [cloudscape-design/components#462]
### Description

Hello 👋

Currently the `header` from `SplitPanel` accepts only a `string`, but would be nice to have it a `React.Node` instead.
There are use cases when it would be good to have a link or some `spaced-between` components on the panel header for information purposes.

As well it would align with other `header` props from other components (that already accepts `React.Node`) like: `ContainerProps`, `AppLayout`, `Table`

Thanks

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/cloudscape-design/components/blob/main/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/cloudscape-design/components/issues) for duplicate requests

**Comment by pan-kot** [comment]
Hello Nichita,

Could you please describe the use cases you have in more detail? We might consider adding more properties to customise the header if needed. Making it a `React.Node` can have an impact on the header's alignment or accessibility.

You can also read more about our split view pattern here: https://cloudscape.design/patterns/resource-management/view/split-view/. Notice the "E" landmark (Details of the selected resources), this could be a good place to put resource-related details such an info link.

**Comment by nichita-pasecinic** [comment]
@pan-kot Sure!

The use case is pretty simple, we've managing a sites (entity) list with a `Table` component. The Site has multiple child entities like resources, services etc., those child/sub entities are visible within Site page itself, but as well the `Resource`/`Service` Pages are implemented as separate pages (with their own list view with split panel).

From the user perspective is easier to see a preview of sites (and see straight on the panel `header` the quick link to the `resources`/`services`  (without actually having to open the panel or open the site information page and only after that to navigate to `resource`/`services`

So what we tried to achieve with the Split panel is the following:

-----
Selected Site Name&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <ins>Resources (12)</ins>  <ins>Services (14)</ins>  ⚙
-----
  

Having the `header` to accept a `React.Node` would expect to add possibility to change this part 
![image](https://user-images.githubusercontent.com/98735377/201075433-d054c932-2653-43a4-8b20-39d08a67fd33.png)

Another workaround is to add some sort of `endActions` prop that could accept a `React.Node` as well, because the need for those actions may vary, and a `React.Node` is a safe choice to let developers implement custom additional functionality over split panel header

Thanks!

**Comment by pan-kot** [comment]
Hello Nichita,

Thank you for sharing the detailed example!

The split panel was designed with the primary objective of “selecting instances at the top and view their details at the bottom (or side)” which is why, when the panel is opened the header displays the "unique resource identifier / number of selected resources". All details are expected to be a part of the split panel content and the split-panel can be opened if the details need to be accessed instantly. Notice, that the split panel also supports side placement where the header takes less space and can't hold much of additional information.

The important information corresponding to the table items can also be placed to the table itself as additional columns.

We did take a note for your request and will use it as a data point when planning future split panel enhancements.

**Comment by nichita-pasecinic** [comment]
Thanks @pan-kot !

**Comment by just-boris** [comment]
Closing as the question has been resolved
